soprano/calculate/nmr/simpson.py

```diff
@@ -116,6 +118,18 @@ def write_spinsys(
 
     nuclei = [str(i) + n for i, n in zip(isotope_list, nuclei)]
 
+    # Ensure obs_nuc appears first in channels
+    if obs_nuc is not None:
+        # Check if obs_nuc is in nuclei
+        if obs_nuc not in nuclei:
+            raise ValueError(
+                f"{obs_nuc} not found in the list of nuclei"
+            )
+        else:
+            channels = [obs_nuc] + [n for n in sorted(set(nuclei)) if n != obs_nuc]
+    else:
+        channels = sorted(set(nuclei))
+
     # Build header
     header = _header_template.format(
```

Comment:
I think we might need to use the `channels` list in the header template - at the moment I don't see where it's used.

Reply:
You're absolutely right. I forgot to update the header part, now it should be ok.